lib/libevent/libevent.nit

```diff
@@ -470,20 +510,33 @@ class ConnectionFactory
 	do
 		var base = listener.base
 		var bev = new NativeBufferEvent.socket(base, fd, bev_opt_close_on_free)
-
-		# Human representation of remote client address
-		var addr_len = 46 # Longest possible IPv6 address + null byte
-		var addr_buf = new CString(addr_len)
-		addr_buf = addrin_to_address(addrin, addr_buf, addr_len)
-		var addr = if addr_buf.address_is_null then
-				"Unknown address"
-			else addr_buf.to_s
-
+		var addr=""
+
+		if not listener isa UnixConnectionListener then
+			# Human representation of remote client address
+			var addr_len = 46 # Longest possible IPv6 address + null byte
+			var addr_buf = new CString(addr_len)
+			addr_buf = addrin_to_address(addrin, addr_buf, addr_len)
+			addr = if addr_buf.address_is_null then
+					"Unknown address"
+				else addr_buf.to_s
+		end
 		var conn = spawn_connection(bev, addr)
 		bev.enable ev_read|ev_write
 		bev.setcb conn
 	end
 
+	# Listen on domain & file for new connection, which will callback `spawn_connection`
+	fun bind_to_unix(hostname: String, file: String): nullable ConnectionListener
```

Comment:
You don't really need a `hostname` when you use unix sockets

Reply:
Indeed... I left it to allow an easy implementation of a common router for both Unix and TCP. If it's not on schedule, I can remove it.